lib/core/src/display/scene/htmlrenderer.rs

```diff
@@ -0,0 +1,104 @@
+use super::{Camera, HTMLScene};
+
+fn eps(value: f32) -> f32 {
+    if value.abs() < 1e-10 {
+        0.0
+    } else {
+        value
+    }
+}
+
+pub struct HTMLRenderer {}
+
+impl HTMLRenderer {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn render(&self, camera: &mut Camera, scene: &HTMLScene) {
+        let (view_width, view_height) = scene.get_dimension();
+        let fov = camera.projection[5] * view_height / 2.0;
+
+        let transform = camera.transform.to_homogeneous().try_inverse().expect("inverse");
+        scene
+            .div
+            .element
+            .style()
+            .set_property("perspective", &format!("{}px", fov))
+            .expect("set perspective");
+        scene
+            .camera
+            .element
+            .style()
+            .set_property(
+                "transform",
+                &format!(
+                    "translateZ({}px)
+             matrix3d({}, {}, {}, {},
+                      {}, {}, {}, {},
+                      {}, {}, {}, {},
+                      {}, {}, {}, {})
+             translate({}px, {}px)",
+                    fov,
+                    eps(transform[0]),
+                    eps(-transform[1]),
+                    eps(transform[2]),
+                    eps(transform[3]),
+                    eps(transform[4]),
+                    eps(-transform[5]),
+                    eps(transform[6]),
+                    eps(transform[7]),
+                    eps(transform[8]),
+                    eps(-transform[9]),
+                    eps(transform[10]),
+                    eps(transform[11]),
+                    eps(transform[12]),
+                    eps(-transform[13]),
+                    eps(transform[14]),
+                    eps(transform[15]),
+                    view_width / 2.0,
+                    view_height / 2.0
+                ),
+            )
+            .expect("set camera transform");
```

Comment:
This is over 30 lines expression. Should be refactored out. Moreover, why are you negating some values? Are we keeping matrices that are different that those needed in CSS3D? If so, why ?

Reply:
Omg, this is horrible. I will beautify this code.

About the css matrices: I had my original implementation prior to the Euler rotation order being different the one in Three.js. I didn't want to lose much time with it, so I ended up taking this https://github.com/mrdoob/three.js/blob/22ed6755399fa180ede84bf18ff6cea0ad66f6c0/examples/js/renderers/CSS3DRenderer.js#L100 as an implementation reference.

This negation looks like a transform needed by CSS3D. And we might don't need that for WebGL, for instance.

It would make sense for us to keep the representation the same between WebGL Rendering objects and HTML Rendering objects.